pkg/detector/preemption.go

```diff
@@ -282,7 +283,32 @@ func (d *ResourceDetector) HandleDeprioritizedPropagationPolicy(oldPolicy policy
 			if err != nil {
 				return
 			}
+
 			klog.Infof("Enqueuing PropagationPolicy(%s/%s) in case of PropagationPolicy(%s/%s) priority changes", potentialPolicy.GetNamespace(), potentialPolicy.GetName(), newPolicy.GetNamespace(), newPolicy.GetName())
+			if sortedPotentialKeys == nil {
+				sortedPotentialKeys = rbt.NewWith(int32DescendingComparator)
+			}
+
+			rawSamePriorityPolicyKeys, _ := sortedPotentialKeys.Get(potentialPolicy.ExplicitPriority())
+			var samePriorityPolicyKeys []util.QueueKey
+			if rawSamePriorityPolicyKeys != nil {
+				samePriorityPolicyKeys = rawSamePriorityPolicyKeys.([]util.QueueKey)
+			}
+
+			samePriorityPolicyKeys = append(samePriorityPolicyKeys, potentialKey)
+			sortedPotentialKeys.Put(potentialPolicy.ExplicitPriority(), samePriorityPolicyKeys)
+		}
+	}
+
+	// No suitable policy to re-queue.
+	if sortedPotentialKeys == nil {
+		return
+	}
+
+	itr := sortedPotentialKeys.Iterator()
+	for itr.Next() {
```

Comment:
Echo from the `Next()` comments:
> If Next() was called for the first time, then it will point the iterator to the first element if it exists.

Is the first element of the black tree the one with the highest priority?

Reply:
that's right.

Reply:
> I remember the red-black tree doesn't guarantee the root is the smallest/largest.

sure

> If Next() was called for the first time, then it will point the iterator to the first element if it exists.

The key lies in the implementation of the iterator. The first element is not the root node but the left-most(min) node.
And through the comparator, you can change the left-most node to the actual maximum value.